airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/steps/gradle.py

```diff
@@ -120,36 +115,55 @@ async def _run(self) -> StepResult:
             # Set RUN_IN_AIRBYTE_CI to tell gradle how to configure its build cache.
             # This is consumed by settings.gradle in the repo root.
             .with_env_variable("RUN_IN_AIRBYTE_CI", "1")
+            # Disable the Ryuk container because it needs privileged docker access which it can't have.
+            .with_env_variable("TESTCONTAINERS_RYUK_DISABLED", "true")
+            # Set the current working directory.
+            .with_workdir("/airbyte")
+        )
+
+        # Mount the whole git repo to update the cache volume contents and build the CDK.
+        with_whole_git_repo = (
+            gradle_container_base
+            # Mount the whole repo.
+            .with_directory("/airbyte", self.context.get_repo_dir("."))
+            # Update the cache in place by executing a gradle task which will update all dependencies and build the CDK.
+            .with_exec(
+                sh_dash_c(
+                    [
+                        # Ensure that the .m2 directory exists.
+                        "mkdir -p /root/.m2",
+                        # Load from the cache volume.
+                        "(rsync -a --stats --mkpath /root/gradle-cache/ /root/.gradle || true)",
+                        # Resolve all dependencies and write their checksums to './gradle/verification-metadata.dryrun.xml'.
+                        self._get_gradle_command("help", "--write-verification-metadata", "sha256", "--dry-run"),
+                        # Build the CDK and publish it to the local maven repository.
+                        self._get_gradle_command(":airbyte-cdk:java:airbyte-cdk:publishSnapshotIfNeeded"),
+                        # Store to the cache volume.
+                        "(rsync -a --stats /root/.gradle/ /root/gradle-cache || true)",
+                    ]
+                )
+            )
+        )
+
+        # Mount only the code needed to build the connector.
+        gradle_container = (
+            gradle_container_base
             # TODO: remove this once we finish the project to boost source-postgres CI performance.
             .with_env_variable("CACHEBUSTER", hacks.get_cachebuster(self.context, self.logger))
-            # Mount the gradle cache volume.
-            # We deliberately don't mount it at $GRADLE_HOME, instead we load it there and store it from there using rsync.
-            # This is because the volume is accessed concurrently by all GradleTask instances.
-            # Hence, why we synchronize the writes by setting the `sharing` parameter to LOCKED.
-            .with_mounted_cache("/root/gradle-cache", self.connector_java_build_cache, sharing=CacheSharingMode.LOCKED)
-            # Mount the parts of the repo which interest us in /airbyte.
-            .with_workdir("/airbyte")
+            # Copy the local maven repository and force evaluation of `with_whole_git_repo` container.
+            .with_directory("/root/.m2", await with_whole_git_repo.directory("/root/.m2"))
```

Comment:
```suggestion
            .with_directory("/root/.m2", await with_whole_git_repo.directory("/root/.m2"))
```
Can we store the `/root/.m2` path in a class constant which would give it a more explicit name like `GRADLE_DEPENDENCIES_PATH`.

Reply:
This is the local maven repository path. Will do.